packages/sourcecred/src/cli/update/utils.js

```diff
@@ -0,0 +1,24 @@
+// @flow
+
+import cloneDeep from "lodash.clonedeep";
+
+// This is a type-friendly way to add optional attributes to a config.
+// Attributes from the `optionals` parameter will only be added if they
+// are not undefined.
+// It is recommended to define T explicitly when using this.
+// Example: `const object = buildObject<MyType>(...);`
```

Comment:
I think some example invocations using the options or even just exclusions would be super useful here. The API is not exactly self-explanatory. Also not entirely sure why this needs to be in the cli/update path. This seems to have some broader application. Could see it in a path having to do with config definitions, honestly.

Reply:
I'll look for another place to put it and will add more documentation.